bee-protocol/src/workers/requester/milestone.rs

```diff
@@ -0,0 +1,272 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    types::metrics::NodeMetrics,
+    workers::{
+        packets::MilestoneRequestPacket, peer::PeerManager, sender::Sender, storage::StorageBackend, MetricsWorker,
+        PeerManagerResWorker,
+    },
+};
+
+use bee_message::milestone::MilestoneIndex;
+use bee_network::PeerId;
+use bee_runtime::{node::Node, shutdown_stream::ShutdownStream, worker::Worker};
+use bee_tangle::{Tangle, TangleWorker};
+
+use async_trait::async_trait;
+use futures::StreamExt;
+use fxhash::FxBuildHasher;
+use log::{debug, info, warn};
+use tokio::{
+    sync::{mpsc, RwLock},
+    time::interval,
+};
+use tokio_stream::wrappers::{IntervalStream, UnboundedReceiverStream};
+
+use std::{
+    any::TypeId,
+    collections::HashMap,
+    convert::Infallible,
+    time::{Duration, Instant},
+};
+
+const RETRY_INTERVAL_MS: u64 = 2500;
+
+pub(crate) async fn request_milestone<B: StorageBackend>(
+    tangle: &Tangle<B>,
+    milestone_requester: &mpsc::UnboundedSender<MilestoneRequesterWorkerEvent>,
+    requested_milestones: &RequestedMilestones,
+    index: MilestoneIndex,
+    to: Option<PeerId>,
```

Comment:
Pedantic: Maybe `from`? Or consider changing the function name to `send_request_milestone`  because otherwise the `to` sounds odd.

Reply:
what about `peer_id`